lib/src/app/cast.js

```diff
@@ -17,59 +19,73 @@ import '../styles/player.css';
  * @param {string} namespace - namespace to listen to
  * @param {PageLevelParameters} [pageLevelParameters]
  */
-export function init(namespace, pageLevelParameters) {
-    var mediaManager = null;
-    var castPlayer = new CastPlayer(namespace, pageLevelParameters);
-
-    logger.setLevel(process.env.LOG_LEVEL);
-    UIManager.setStatusCast(UIManager.status.LOADING);
+export function init (namespace, pageLevelParameters) {
+  logger.setLevel(process.env.LOG_LEVEL);
+  UIManager.setStatusCast(UIManager.status.LOADING);
 
-    // create a new instance of the Media Manager to interact with the sender app request
-    mediaManager = new cast.receiver.MediaManager(castPlayer);
+  const castPlayer = new CastPlayer(namespace, pageLevelParameters);
+  const mediaManager = new cast.receiver.MediaManager(castPlayer);
 
-    // backup of the original onLoad event handler to use it after our custom logic
+  /**
+   * Exotic class method extending by Google devs
+   * @see https://developers.google.com/cast/docs/custom_receiver (section Media)
+   * onLoad assigns a new implementation of the onLoad event that can set the
+   * needed information from the incoming event
+   * @param {cast.receiver.MediaManager.Event} event
+   */
+  mediaManager.onLoad = (() => {
     mediaManager.origOnLoad = mediaManager.onLoad;
-
-    /**
-     * onLoad assign a new implementation of the onLoad event that can sets the needed information from the
-     * incoming event
-     * @param {cast.receiver.MediaManager.Event} event
-     */
-    mediaManager.onLoad = function(event) {
-        logger.info('MediaManager:onLoad', event);
-        if (event && event.data && event.data.autoplay) {
-            Timer.setIdle();
-        }
-        castPlayer.setAsset(event.data.media.customData);
-        if (castPlayer.OOPlayer != null) {
-            castPlayer.notifySenders(castPlayer.OOPlayer.getState());
+    return (event) => {
+      logger.info('MediaManager:onLoad', event);
+      const isAutoplayDesired = event && event.data && event.data.autoplay;
+      if (isAutoplayDesired) {
+        Timer.setIdle();
+      }
+      castPlayer.setAsset(event.data.media.customData);
+      if (castPlayer.OOPlayer != null) {
+        const state = castPlayer.OOPlayer.getState();
+        castPlayer.notifySenders(state);
+        if (isAutoplayDesired) {
```

Comment:
The description for this PR said that there were no logic changes. This if statement looks to be a logic change, so just double checking if this was intentional.

Reply:
Sorry, I've branched not from master occasionally. Rolled this back.